tools/scripts/script.ps1

```diff
@@ -0,0 +1,242 @@
+        $osPlatform = "Windows"
+        $destinationArchiveFile = Join-Path $tempFilesDirectory "AIShell.zip"
+    } 
+    elseif ($IsMacOS) {
+        $osPlatform = "MacOS"
+        $destinationArchiveFile = Join-Path $tempFilesDirectory "AIShell.tar.gz"
+    }
+    Write-Host "[Downloading the latest AIShell for $osPlatform $osArchitecture to $destinationArchiveFile ...]`n" -ForegroundColor Green
+    Invoke-WebRequest -Uri $buildUrl -OutFile $destinationArchiveFile
+    
+    # Unzip AIShell
+    Write-Host "[Unzipping AIShell to $destinationDirectory ...]" -ForegroundColor Green
+    Unblock-File -Path $destinationArchiveFile
+    Expand-Archive -Path $destinationArchiveFile -DestinationPath $destinationDirectory -Force
+    if ($IsMacOS) {
+        $aishPath = Join-Path $destinationArchiveFile "aish"
+        chmod +x $aishPath
+    }
+}
+
+function Remove-AIShellFromLocalDir {
+    $DestinationPath = Get-InstallDirectoryForCurrentOS
+    if (Test-Path $DestinationPath) {
+            Write-Host "[Removing AIShell from $DestinationPath]" -ForegroundColor Green
+            Remove-Item -Path $DestinationPath -Recurse -Force
+            Write-Host "[AIShell removed from $DestinationPath]" -ForegroundColor Green
+    } 
+    else {
+        Write-Host "[AIShell cannot be found at $DestinationPath, skip removing]" -ForegroundColor Yellow
+    }
+}
+
+function Register-AIShellModule {
+    Write-Host "[Installing AIShell Module]" -ForegroundColor Green
+    $aishModulePath = Join-Path $(Get-InstallDirectoryForCurrentOS) "temp"
+    $repoName = "tempRepositoryForAIShell"
+    Register-PSRepository -Name $repoName -SourceLocation $aishModulePath -InstallationPolicy Trusted
+    Install-Module -Name "AIShell" -Repository $repoName -AllowClobber
+    Unregister-PSRepository -Name $repoName
+    Remove-Item -Path $aishModulePath -Recurse -Force
+    Write-Host "[AIShell Module installed]" -ForegroundColor Green
+}
+
+function Unregister-AIShellModule {
+    if (Get-InstalledModule -Name "AIShell") {
+        Write-Host "[Uninstalling AIShell Module]" -ForegroundColor Green
+        Uninstall-Module -Name "AIShell" -Force
+        Write-Host "[AIShell Module uninstalled]" -ForegroundColor Green
+    } 
+    else {
+        Write-Host "[AIShell Module cannot be found, skip uninstalling]" -ForegroundColor Yellow
+    }
+}
+
+function Test-AIShellExistInPath {
+    $aishPath =  Get-InstallDirectoryForCurrentOS
+    return ($env:PATH -like "*$aishPath*")
+}
+
+function Add-AIShellToPath {
+    if (Test-AIShellExistInPath) {
+        Write-Host "[AIShell was already added in PATH]" -ForegroundColor Green
+    } else {
+        Write-Host "[Adding AIShell to PATH]" -ForegroundColor Green
+        $processPath = $env:PATH
+        if (!$processPath.EndsWith("$")) {
+            $processPath += ";"
+        }
+        $aishPath =  Get-InstallDirectoryForCurrentOS
+        [Environment]::SetEnvironmentVariable("PATH", $processPath + $aishPath, "Process")
```

Comment:
Only adding to the process-scope Path doesn't persist the setting. Running `aish` won't work from a different PowerShell session.
On Unix platform, we don't manipulate the Path environment variable directly, but instead create a symbolic link in a well-known folder that's already in the Path. Think about how Az CLI does it when installing on mac and Linux.